Mathlib/Util/AssertExists.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot, Scott Morrison
+-/
+import Lean
+
+/-!
+# User commands for assert the (non-)existence of declaration or instances.
+
+These commands are used to enforce the independence of different parts of mathlib.
+
+## TODO
+
+Potentially after the port reimplement the mathlib 3 linters to check that declarations asserted
+about do eventually exist.
+
+Implement `assert_instance` and `assert_no_instance`
+-/
+
+section
+open Lean Elab Meta
+
+/--
+`assert_exists n` is a user command that asserts that a declaration named `n` exists
+in the current import scope.
+
+Be careful to use names (e.g. `Rat`) rather than notations (e.g. `ℚ`).
+-/
+elab "assert_exists " n:ident : command => do
+  let _ ← resolveGlobalConstNoOverloadWithInfo n
```

Comment:
Is the reason for using this spelling rather than
```suggestion
  let _ ← resolveGlobalConstNoOverloadWithInfos n
```
to enforce there should be exactly one decl with this name?

Reply:
added a comment